backends/ris/decoder.go

```diff
@@ -196,23 +196,27 @@ func mapRecord(r Record, p *models.Publication) {
 				p.Type = "miscellaneous"
 				p.MiscellaneousType = "other"
 			}
-		case "AF", "AU":
-			// give preference to AF over AU
-			if k == "AU" && p.Author != nil {
-				continue
+		case "AF":
+			for _, val := range v {
+				p.Author = append(p.Author, extractContributor(val))
 			}
-			if k == "AF" {
-				p.Author = nil
+		case "AU":
+			// give preference to AF over AU
+			if _, ok := r["AF"]; !ok {
+				for _, val := range v {
+					p.Author = append(p.Author, extractContributor(val))
+				}
```

Comment:
Make sure to carefully review and test this. I've refactored this and tested somewhat but not entirely sure if the new code is completely equivalent to the old code.

Reply:
looks good to me